core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java

```diff
@@ -11,61 +11,55 @@
 
 package alluxio.proxy.s3;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
 import com.google.protobuf.ByteString;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import javax.annotation.Nullable;
+import javax.xml.bind.annotation.XmlTransient;
 
 /**
  * Parsed version of the S3 tagging XML.
  * Used by the following S3 API actions:
  * - GetBucketTagging, GetObjectTagging, PutBucketTagging, PutObjectTagging
  */
 @JacksonXmlRootElement(localName = "Tagging")
-public class TaggingData implements Serializable {
-  public static final long serialVersionUID = 1L;
```

Comment:
I recently learned that I wasn't serializing this as an XML, but rather as a plain Java object (d'oh). So I switched the implementations in this PR.